public/dorequest.php

```diff
@@ -114,6 +120,52 @@ function DoRequestError(string $error, ?int $status = 200, ?string $code = null)
     return DoRequestError("You do not have permission to do that.", 403, 'access_denied');
 }
 
+/*
+ * It is possible for some calls to be made on behalf of another user.
+ * This is only currently supported for "Standalone" integrations.
+ * NOTE: "awardachievements" is not included here because it accepts an array of
+ * achievement ID values. It doesn't use the generic `delegateUserAction()` function.
+ */
+$allowsGenericDelegation = [
+    "awardachievement",
+    "ping",
+    "startsession",
+];
+if (
+    in_array($requestType, $allowsGenericDelegation)
+    && $delegateTo !== null
+    && ($gameID || $achievementID)
+) {
+    if (request()->method() !== 'POST') {
+        return DoRequestError('Access denied.', 403, 'access_denied');
+    }
+
+    $foundDelegateToUser = User::firstWhere('User', $delegateTo);
+    if (!$foundDelegateToUser) {
+        return DoRequestError("The target user couldn't be found.", 403, 'access_denied');
```

Comment:
Is 403 the best response for this? Maybe 404/not_found?

Same comment for missing game/achievements below.

Reply:
404 is probably a better fit. Updated in latest.